src/index.js

```diff
@@ -0,0 +1,195 @@
+                setTimeout(() => {
+                    resolve(getWithRetry(url, queryParams, attempt + 1));
+                }, 600 * attempt);
+            });
+        }
+        return response;
+    })
+};
+
+const findLatitudeAndLongitude = (query) => {
+    const url = "http://localhost:5000/location";
+    return getWithRetry(url, { q: query, format: 'json' }).then((response) => {
+        if (!response.data) {
+            return null;
+        }
+        const { lat: latitude, lon: longitude } = response.data[0];
+        return { latitude, longitude };
+    });
+};
+const findWeatherLatLon = (lat, lon) => {
+    const url = "http://localhost:5000/weather";
+
+    return getWithRetry(url, { lat, lon }).then((response) => {
+        if (!response.data) {
+            return null;
+        }
+        const tempK = response.data.main.temp;
+        const tempC = Math.round(tempK - 273.15);
+        const tempF = Math.round((tempC * 9) / 5 + 32);
+        return { tempF, tempC };
+    });
+};
+
+currentTempButton.addEventListener("click", () => {
+    const city = cityNameElement.textContent.trim();
+    if (!city) return;
+
+    findLatitudeAndLongitude(city)
+        .then(({ latitude, longitude }) => findWeatherLatLon(latitude, longitude))
+        .then((result) => {
+            tempF = result.tempF;
+            updateTempColor();
+        });
+});
+
+// wave 5 //
+function updateSkyEmoji(selectedSky) {
+    const skySection = document.querySelector(".sky__section");
+    const emojiMap = {
+        sunny: "☀️",
+        cloudy: "☁️",
+        rainy: "🌧",
+        snowy: "❄️",
+        select_one: "👈🏼"
+    };
+    skySection.dataset.emoji = emojiMap[selectedSky] || "";
+}
+
+function selectTheSkyDropdown() {
+    skySelect.addEventListener("change", () => {
+        const selectedSky = skySelect.value;
+        updateSkyEmoji(selectedSky);
+
+        if (selectedSky === 'sunny') {
+            sky.textContent = "☁️ ☁️ ☁️ ☀️ ☁️ ☁️";
+        } else if (selectedSky === 'cloudy') {
+            sky.textContent = "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️";
+        } else if (selectedSky === 'rainy') {
+            sky.textContent = "🌧🌈🌧🌧💧🌧🌦🌧💧🌧🌧";
+        } else if (selectedSky === 'snowy') {
+            sky.textContent = "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨";
+        } else {
+            sky.textContent = "";
+        }
+        updateSkyEmoji(selectedSky); 
+    });
+}
+selectTheSkyDropdown();
+
+// wave 6: //
+function resetCityName() {
+    cityNameReset.addEventListener("click", () => {
+        cityNameInput.value = "";
+        headerCityName.textContent = "";
+        tempF = 0;
+        updateTempColor();
+        sky.textContent = "";
+        skySelect.value = "select_one";
+        updateSkyEmoji("select_one");
+    });
+}
+
+resetCityName();
+document.addEventListener("DOMContentLoaded", () => {
+    skySelect.value = "select_one";
+    sky.textContent = "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨";
+    updateSkyEmoji("select_one");
+});
```

Comment:
Okay, I see that you used `DOMContentLoaded`! Nice! I think you just need to be more consistent in your use of it. Specifically, using it for all events registration logic and function invokes you need to run without being triggered by users when the DOM is first loaded.